src/shared/bus-polkit.c

```diff
@@ -399,19 +507,13 @@ int bus_verify_polkit_async(
                 return -ENOMEM;
         }
 
-        r = hashmap_put(*registry, call, q);
-        if (r < 0) {
-                async_polkit_query_free(q);
-                return r;
-        }
-
-        q->registry = *registry;
+        LIST_PREPEND(item, qs->items, q);
 
         r = sd_bus_call_async(call->bus, &q->slot, pk, async_polkit_callback, q, 0);
-        if (r < 0) {
-                async_polkit_query_free(q);
```

Comment:
Why this is dropped?? `q` seems to be leaked.

Reply:
It's not leaked. At this point, `q` is already managed by `qs`.

Reply:
Ah, then, I'd like to move LIST_PREPEND in the above after the bus call.